src/main/java/net/mcreator/ui/dialogs/AIConditionEditor.java

```diff
@@ -54,9 +54,15 @@
 			continueCondition.setSelectedProcedure(data[1]);
 		}
 
-		JOptionPane pane = new JOptionPane(PanelUtils.gridElements(1, 2, 10, 10, startCondition, continueCondition));
-		JDialog dialog = pane.createDialog(parent, L10N.t("dialog.ai_condition.panel_name"));
-		dialog.setVisible(true);
+		MCreatorDialog window = new MCreatorDialog(parent, L10N.t("dialog.ai_condition.panel_name"));

```

Comment:
Clicking "x" to save changes is not so intuitive so add OK button on the bottom as it was before.

Reply:
I added OK & Cancel buttons to the dialog. If one re-selects conditions and clicks OK, procedures are updated, meanwhile using Cancel/Close button closes the dialog and ignores changes. Does it sound good?

Reply:
I added OK & Cancel buttons to the dialog. If one re-selects conditions and clicks OK, procedures are updated, meanwhile using Cancel/Close button ignores changes made. Every button closes the dialog when clicked. Does it sound good?

Reply:
Sounds good ;)